### PR TITLE
WIP: Support the API tester

### PR DESCRIPTION
**Pull request by C0DK**
Status: 
Latest is allmost installed as a middleware. however some typing is fucking me over.

Then i also need to add/change existing endpoints, and fix the json conversion.
I'll look at it later. have to go to IBM now.

closes: #27

### CODE REVIEW

File: src/WebApplication/Startup.cs [hunk 1]
```diff
@@ -55,6 +56,15 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment environment)
...
             }
+            app.Use(async (context, next) =>
+            {
+                // Do work that doesn't write to the Response.
+                if(context.Request.Query.TryGetValue("latest", out string latestString) && int.TryParse(latestString, out int latest)) {
+                    TestingUtils.SetLatest(latest);
+                }
+                await next.Invoke();
+                // Do logging or other work that doesn't write to the Response.
+            });
```
Comment (Trivivium) [inline src/WebApplication/Startup.cs]: This should property be placed after the `UseAuthentication()` middleware to ensure only authenticated users can run this endpoint.

File: schema.sql [hunk 2]
```diff
@@ -20,3 +20,16 @@ create table message (
...
+ * that can then be queried to see that requests
+ * are actually handled. This row is used for that -
+ * it is essentially a singleton. I'm sorry -
+ * it is part of the requirements.
+*/
+drop table if exists latest;
+create table latest (
+  id integer primary key,
+  latest integer
+);
```
Comment (Trivivium) [inline schema.sql]: If we're to store the value across requests we might be able to get away with an im-memory cache. Unless of course the simulator expects the value to always be there after the initial request.

File: src/WebApplication/Startup.cs (hunk 1 above)
Comment (C0DK) [reply src/WebApplication/Startup.cs]: cool. thanks. will move, when i make it work. right now i have typing issues

File: schema.sql (hunk 2 above)
Comment (C0DK) [reply schema.sql]: i wanted this just to be sure

**Pull request merged**